### PR TITLE
Add/expected lite

### PR DESCRIPTION
**Pull request by flexferrum**
Specify library name and version:  **expected-lite/0.3.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by flexferrum** [comment]
According to the documentation: https://github.com/martinmoene/expected-lite
the library can't be compiled with standards lower than C++11:
> expected lite is a single-file header-only library for objects that either represent a valid value or an error that you can pass by value. It is intended for use with C++11 and later.

So, is there a way to limit the set of standards for this library?

File: recipes/expected-lite/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 2.8.11)
+project(test_package CXX)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```
Comment (uilianries) [inline recipes/expected-lite/all/test_package/CMakeLists.txt]:
```suggestion
target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
set_target_properties(${PROJECT_NAME} PROPERTIES CXX_STANDARD 11)
```

Comment (uilianries) [reply recipes/expected-lite/all/test_package/CMakeLists.txt]: Just add C++11 support to your test package

Comment (flexferrum) [reply recipes/expected-lite/all/test_package/CMakeLists.txt]: Thanks. Sure.

File: recipes/expected-lite/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+class ExpectedLite(ConanFile):
+    name = "expected-lite"
+    
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/martinmoene/expected-lite"
+    description = "expected lite - Expected objects in C++11 and later in a single-file header-only library"
+    author = "Martin Moene"
+    topics = ("conan", "cpp11", "cpp14", "cpp17", "expected", "expected-implementations")
+    license = "BSL-1.0"
+
```
Comment (uilianries) [inline recipes/expected-lite/all/conanfile.py]:
```suggestion
    no_copy_source = True
```
As you dont need to build anything, you don't need to copy to build folder

File: recipes/expected-lite/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import os
+from conans import ConanFile, CMake, tools
```
Comment (uilianries) [inline recipes/expected-lite/all/conanfile.py]:
```suggestion
from conans import ConanFile, tools
```
No usage for cmake

**Pull request merged**